transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -686,10 +686,16 @@ private boolean recvmsg(EpollRecvByteAllocatorHandle allocHandle,
             NativeDatagramPacketArray.NativeDatagramPacket msg = array.packets()[0];
 
             int bytesReceived = socket.recvmsg(msg);
-            if (!msg.hasSender()) {
+            if (bytesReceived < 0 && Errors.ioResult("recvmsg(...)", bytesReceived) == 0) {
```

Comment:
I _believe_ the result of `socket.recvmsg` has already called `Errors.ioResult(..)` via `Native.recvmsg`. Same with recvmmsg.

Reply:
brainfart... Fixed it